src/com/github/igotyou/FactoryMod/managers/CompactorManager.java

```diff
@@ -22,25 +22,25 @@ public CompactorManager(FactoryModPlugin plugin) {
     @Override
     public InteractionResponse createFactory(Location factoryLocation,
             Location inventoryLocation, Location powerLocation) {
-    	CompactorProperties compactorProperties = plugin.getCompactorProperties();
-		
-		if (!factoryExistsAt(factoryLocation))
-		{
-			Block inventoryBlock = inventoryLocation.getBlock();
-			Chest chest = (Chest) inventoryBlock.getState();
-			Inventory chestInventory = chest.getInventory();
-			ItemList<NamedItemStack> inputs = compactorProperties.getConstructionMaterials();
-			boolean hasMaterials = inputs.allIn(chestInventory);
-			if (hasMaterials)
-			{
+        CompactorProperties compactorProperties = plugin.getCompactorProperties();
+    	
+        if (!factoryExistsAt(factoryLocation)) {
+            Block inventoryBlock = inventoryLocation.getBlock();
+            Chest chest = (Chest) inventoryBlock.getState();
+            Inventory chestInventory = chest.getInventory();
+            ItemList<NamedItemStack> inputs = compactorProperties.getConstructionMaterials();
+
+			if (inputs.exactlyIn(chestInventory)) {
 				Compactor production = new Compactor(factoryLocation, inventoryLocation, powerLocation, false, plugin.getCompactorProperties());
-				if (compactorProperties.getConstructionMaterials().removeFrom(production.getInventory()))
-				{
-					addFactory(production);
+				inputs.removeFrom(production.getInventory());
```

Comment:
Note to self: Now that super.addFactory() is a boolean function, removal of recipe ingredients should happen *after* that method returns true. E.g. move this down two lines. Same for Repair Factory.